### PR TITLE
feat(datastore): Add non-model type support for amplify-flutter

### PR DESCRIPTION
**Pull request by HuiSF**
*Issue #, if available:* amplify-flutter [#260](https://github.com/aws-amplify/amplify-flutter/issues/260)

*Description of changes:*

* Make `ModelSchemaRegistry` support registering `CustomType` schemas
* Make AppSync able to process GraphQL request for `SerializedModel` nesting `SerializedCustomType`
* Make DataStore able to read/write with `SerializedCustomType` for amplify-flutter

**Tasks**

- [x] Rename `ModelSchemaRegistry` to `SchemaRegistry`
- [x] Implement `CustomTypeSchema`, `SerializedCustomType` and related classes
- [x] Implement additional logic to support serializing and deserializing with `SerializedCustomType` with amplify-flutter use cases
- [x] Add unit tests for code changes in `aws-api-appsync`
- [x] Add unit tests for code changes in `aws-datastore`
- [x] Add unit tests for code changes in `core`
----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: core/src/main/java/com/amplifyframework/core/model/SchemaRegistry.java [hunk 1]
```diff
@@ -104,19 +140,33 @@ public synchronized ModelSchema getModelSchemaForModelClass(@NonNull String clas
...
      */
     @NonNull
-    public static synchronized ModelSchemaRegistry instance() {
-        return new ModelSchemaRegistry();
+    public static synchronized SchemaRegistry instance() {
+        if (SchemaRegistry.instance == null) {
+            SchemaRegistry.instance = new SchemaRegistry();
+        }
+
+        return SchemaRegistry.instance;
```
Comment (HuiSF) [inline core/src/main/java/com/amplifyframework/core/model/SchemaRegistry.java:162]: From [this use case](https://github.com/aws-amplify/amplify-android/blob/c241e5128f28bd4a69d501226458cbf86fa10c9b/aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java#L368-L369) of `SchemaRegistry`, it looks like `SchemaRegistry` should be a singleton. The original code didn't work and didn't break either as the original logic wasn't really depending on retrieving a valid model schema.

**Review by poojamat — APPROVED** [review]

File: aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java [hunk 2]
```diff
@@ -28,7 +28,7 @@
 import com.amplifyframework.core.model.ModelAssociation;
 import com.amplifyframework.core.model.ModelField;
 import com.amplifyframework.core.model.ModelSchema;
-import com.amplifyframework.core.model.ModelSchemaRegistry;
```
[the remark this answers is not in the document]
Comment (HuiSF) [reply aws-api-appsync/src/main/java/com/amplifyframework/api/aws/SelectionSet.java:31]: Thanks for asking :)
`ModelSchemaRegistry` not holds `CustomTypeSchema` as well, the original name was leading by `Model` which could be confusing, so updated to just `SchemaRegistry`.

File: core/src/main/java/com/amplifyframework/core/model/SerializedModel.java [hunk 3]
```diff
@@ -46,9 +48,10 @@ private SerializedModel(
 
     /**
      * Creates a SerializedModel from a generated Java Model object.
-     * @param model Model object
+     *
+     * @param model       Model object
      * @param modelSchema schema for the Model object
-     * @param <T> type of the Model object.
+     * @param <T>         type of the Model object.
```
Comment (Jordan-Nelson) [inline core/src/main/java/com/amplifyframework/core/model/SerializedModel.java:54]: *nit:* some of the formatting looks like it needs to be corrected

Comment (HuiSF) [reply core/src/main/java/com/amplifyframework/core/model/SerializedModel.java:54]: This formatting is automatically added by the IDE I think, which align parameter description block with each other for readability.

**Review by Jordan-Nelson — APPROVED** [review]
LGTM

**Pull request merged**